runtime/battery-station/src/integration_tests/xcm/setup.rs

```diff
@@ -0,0 +1,210 @@
+    VersionedMultiLocation,
+};
+use zeitgeist_primitives::types::Asset;
+
+pub(super) struct ExtBuilder {
+    balances: Vec<(AccountId, CurrencyId, Balance)>,
+    parachain_id: u32,
+}
+
+impl Default for ExtBuilder {
+    fn default() -> Self {
+        Self { balances: vec![], parachain_id: battery_station::ID }
+    }
+}
+
+impl ExtBuilder {
+    pub fn balances(mut self, balances: Vec<(AccountId, CurrencyId, Balance)>) -> Self {
+        self.balances = balances;
+        self
+    }
+
+    pub fn parachain_id(mut self, parachain_id: u32) -> Self {
+        self.parachain_id = parachain_id;
+        self
+    }
+
+    pub fn build(self) -> sp_io::TestExternalities {
+        let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+        let native_currency_id = CurrencyId::Ztg;
+        pallet_balances::GenesisConfig::<Runtime> {
+            balances: self
+                .balances
+                .clone()
+                .into_iter()
+                .filter(|(_, currency_id, _)| *currency_id == native_currency_id)
+                .map(|(account_id, _, initial_balance)| (account_id, initial_balance))
+                .collect::<Vec<_>>(),
+        }
+        .assimilate_storage(&mut t)
+        .unwrap();
+
+        orml_tokens::GenesisConfig::<Runtime> {
+            balances: self
+                .balances
+                .into_iter()
+                .filter(|(_, currency_id, _)| *currency_id != native_currency_id)
+                .collect::<Vec<_>>(),
+        }
+        .assimilate_storage(&mut t)
+        .unwrap();
+
+        <parachain_info::GenesisConfig as GenesisBuild<Runtime>>::assimilate_storage(
+            &parachain_info::GenesisConfig { parachain_id: self.parachain_id.into() },
+            &mut t,
+        )
+        .unwrap();
+
+        <pallet_xcm::GenesisConfig as GenesisBuild<Runtime>>::assimilate_storage(
+            &pallet_xcm::GenesisConfig { safe_xcm_version: Some(2) },
+            &mut t,
+        )
+        .unwrap();
+
+        let mut ext = sp_io::TestExternalities::new(t);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+}
+
+/// Accounts
+pub const ALICE: [u8; 32] = [4u8; 32];
+pub const BOB: [u8; 32] = [5u8; 32];
+
+/// A PARA ID used for a sibling parachain.
+/// It must be one that doesn't collide with any other in use.
+pub const PARA_ID_SIBLING: u32 = 3000;
+
+/// IDs that are used to represent tokens from other chains
+pub const FOREIGN_ZTG_ID: Asset<u128> = CurrencyId::ForeignAsset(0);
+pub const FOREIGN_PARENT_ID: Asset<u128> = CurrencyId::ForeignAsset(1);
+pub const FOREIGN_SIBLING_ID: Asset<u128> = CurrencyId::ForeignAsset(2);
+
+// Multilocations that are used to represent tokens from other chains
+#[inline]
+pub(super) fn foreign_ztg_multilocation() -> MultiLocation {
+    MultiLocation::new(1, X2(Parachain(battery_station::ID), general_key(battery_station::KEY)))
```

Comment:
why use general_key here? Also when we use Parachain() junction it is meant for parachain belonging to and operated by the context, then how is this a foreign ztg multi location ?

Reply:
It was my fault in understanding. I think this is fine.